src/connectors/TokenConnectorLogic.sol

```diff
@@ -91,8 +78,16 @@ abstract contract TokenConnectorLogic is IBridgeConnector {
             finalizedState = state;
             state = new TokenState(epoch_to_finalize + 1);
         }
+        Transfer[] memory epochTransfers = finalizedState.getTransfers();
+        if (epochTransfers.length > 0) {
+            uint256 settlementFeesToForward = bridge.settlementFee() * epochTransfers.length;
+            (bool success,) = address(bridge).call{value: settlementFeesToForward}("");
```

Comment:
@kstdl maybe we could add a balance check here that reverts with a specific error instead of looking through the `TransferFailed` reverts if the connector can't pay.